### PR TITLE
Add Sphinx autodoc tooling

### PR DESCRIPTION
**Pull request by pwilkins**
Add Sphinx documentation tools to generate API docs
from code docstrings. Uses Google doc style through Napoleon
and the Sphinx Bootstrap theme.

See PT story here: https://www.pivotaltracker.com/story/show/95632294

Closes Github issue #55 - https://github.com/mitodl/lore/issues/55

### CODE REVIEW

**Comment by pwilkins** [comment]
@carsongee @ShawnMilo @pdpinch @noisecapella @giocalitri This is a first pass at API docs.  I'd welcome your comments and suggestions for improvement.

**Comment by ShawnMilo** [comment]
It looks pretty straightforward to me -- mostly Sphinx boilerplate. Is there a one-step command to build the documentation with `docker-compose`?

**Comment by pwilkins** [comment]
@ShawnMilo - @carsongee suggested creating a TOX env task to build the doc.  I don't know if he wanted that to run by default or only explicitly.

**Comment by carsongee** [comment]
There are pep8 violations in the conf, either fix or disable in `pytest.ini`.  Also, there is a tox example right in the docs for sphinx: http://tox.readthedocs.org/en/latest/example/general.html#integrating-sphinx-documentation-checks

File: Dockerfile [hunk 1]
```diff
@@ -16,8 +17,10 @@ RUN pip install pip --upgrade
 # Install project packages
 RUN pip install -r requirements.txt
 RUN pip install -r test_requirements.txt
+RUN pip install -r doc_requirements.txt
 RUN pip3 install -r requirements.txt
 RUN pip3 install -r test_requirements.txt
+RUN pip3 install -r doc_requirements.txt
```
Comment (carsongee) [inline Dockerfile]: We just need to choose one or the other, no reason to slow the build for testing in both python languages.  I think readthedocs uses 2, so we can just do it there.

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+#texinfo_appendices = []
+
+# If false, no module index is generated.
+#texinfo_domain_indices = True
+
+# How to display URL addresses: 'footnote', 'no', or 'inline'.
+#texinfo_show_urls = 'footnote'
+
+# If true, do not generate a @detailmenu in the "Top" node's menu.
+#texinfo_no_detailmenu = False
```
Comment (carsongee) [inline docs/conf.py]: I would just nuke everything that is commented out.

File: Dockerfile (hunk 1 above)
Comment (pwilkins) [inline Dockerfile]: @carsongee Do I need to change tox.ini to explicitly run the docs only in py27?

**Comment by carsongee** [comment]
Looks like the rtfd build is failing on the django conf stuff: https://readthedocs.org/builds/lore/2886899/

**Comment by pwilkins** [comment]
ReadTheDocs was failing because it initially tried to build docs from the master branch.  I switched it to the proper branch for the docs.

File: doc_requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+sphinx>=1.3.1
+sphinx_bootstrap_theme>=0.4.5
+sphinxcontrib-napoleon>=0.3.5
```
Comment (carsongee) [inline doc_requirements.txt:4]: I think you want to add `-r requirements.txt` here and give rtfd this as the requirements to install.  That is how we've done it in the past.   That should simplify the tox build a bit too I think

**Comment by carsongee** [comment]
Woot, this looks great! :loud_sound:

**Pull request merged**